library/std/src/sys/pal/hermit/fd.rs

```diff
@@ -48,6 +48,11 @@ impl FileDesc {
     pub fn set_nonblocking(&self, _nonblocking: bool) -> io::Result<()> {
         unsupported()
     }
+
+    pub fn fstat(&self, stat: *mut abi::stat) -> io::Result<()> {
+        cvt(unsafe { abi::fstat(self.fd.as_raw_fd(), stat) })?;
```

Comment:
...Why did I have to trace through three different directories just to discover that this is referring to the `hermit_abi` crate? That is so confusing.

Reply:
Excuse me for my late response. I am a little bit confused. Why do you have trace three different directories? Do you mean I should move `fstat` from from `fd.rs` to `fs.rs`?

Reply:
Or do you want that rename `abi` to `hermit-abi`? Maybe it is easier to understand...

Reply:
Ah, I was a bit confused by the levels of indirection in general, but yes, I think probably using `hermit_abi` might be a bit clearer.

Reply:
Ok, I would prefer to create an additional PR for these changes because I have to touch every file in `library/std/src/sys/pal/hermit/`. In addition, it doesn't belong to this PR.